lib/acts_as_paranoid/core.rb

```diff
@@ -2,6 +2,12 @@ module ActsAsParanoid
   module Core
     def self.included(base)
       base.extend ClassMethods
+      if ActiveRecord::VERSION::MAJOR > 5 || (ActiveRecord::VERSION::MAJOR == 5 && (ActiveRecord::VERSION::MINOR >= 2 || (ActiveRecord::VERSION::MINOR == 2 && ActiveRecord::VERSION::TINY >= 1)))
+        base.send(:alias_method, :remember_transaction_record_state_without_paranoid, :remember_transaction_record_state)
+        base.send(:alias_method, :remember_transaction_record_state, :remember_transaction_record_state_with_paranoid)
+      end
+      base.send(:alias_method, :force_clear_transaction_record_state_without_paranoid, :force_clear_transaction_record_state)
```

Comment:
You can avoid the `base.send ...` by using `class << base`, as done in `lib/acts_as_paranoid/associations.rb`

Reply:
It is not that simple because in your example, you redefined belongs_to which is a class method and here we are dealing with instance methods. We could use a class_eval however.